corehq/apps/accounting/models.py

```diff
@@ -1230,6 +1232,13 @@ def send_ending_reminder_email(self):
                     'email': email,
                 })
 
+    def set_billing_account_entry_point(self):
+        no_current_entry_point = self.account.entry_point == EntryPoint.NOT_SET
+        self_serve = self.service_type == SubscriptionType.SELF_SERVICE
+        if (no_current_entry_point and self_serve and not self.is_trial):
+            self.account.entry_point = EntryPoint.SELF_STARTED
+            self.account.save()
```

Comment:
Good point. Do you have suggestions on how else could I have done this while ensuring that the account gets saved once `entry_point` has been updated?